api/lib/models/project/project-data-mapper-impl.ts

```diff
@@ -0,0 +1,236 @@
+    }
+    if (opts && opts.count) {
+      queryBuilder = queryBuilder.limit(opts.count);
+    }
+    if (opts && opts.byHackathon) {
+      queryBuilder = queryBuilder
+        .where(
+          'hackathon = ?',
+          await (opts && opts.hackathon ?
+          Promise.resolve(opts.hackathon) :
+          this.activeHackathonDataMapper.activeHackathon
+            .pipe(map(hackathon => hackathon.uid))
+            .toPromise()),
+        );
+    }
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<Project>(query.text, query.values, { cache: true },
+      ))
+      .pipe(map((projects: Project[]) => ({ result: 'Success', data: projects })),
+      )
+      .toPromise();
+  }
+
+  public async getByUser(uid: UidType, opts?: IUowOpts): Promise<IDbResult<Project>> {
+    let queryBuilder = squel.select({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.teamTableName, 'pt')
+      .field('pl.projectName')
+      .field('pt.*')
+      .field('ta.tableNumber')
+      .field('cl.*')
+      .join(this.tableName, 'pl', 'pl.projectID=pl.projectID')
+      .join('PROJECT_CATEGORIES', 'pc', 'pc.projectID = pt.projectID')
+      .join('CATEGORY_LIST', 'cl', 'cl.uid = pc.categoryID ')
+      .left_join('TABLE_ASSIGNMENTS', 'ta', 'ta.projectID=pl.project.ID')
+      .where('pt.userID = ?', uid);
+    if (opts && opts.byHackathon) {
+      queryBuilder = queryBuilder
+        .where(
+          'hackathon = ?',
+          await (opts && opts.hackathon ?
+          Promise.resolve(opts.hackathon) :
+          this.activeHackathonDataMapper.activeHackathon
+            .pipe(map(hackathon => hackathon.uid))
+            .toPromise()),
+        );
+    }
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<Project>(query.text, query.values, { cache: true },
+      ))
+      .pipe(map((project: Project[]) => ({ result: 'Success', data: project[0] })),
+      )
+      .toPromise();
+  }
+
+  public async getCount(opts?: IUowOpts): Promise<IDbResult<number>> {
+    let queryBuilder = squel.select({ autoQuoteTableNames: true, autoQuoteFieldNames: false })
+      .from(this.tableName)
+      .field(`COUNT(${this.pkColumnName})`, 'project_count');
+    if (opts && opts.byHackathon) {
+      queryBuilder = queryBuilder
+        .where(
+          'hackathon = ?',
+          await (opts.hackathon ?
+            Promise.resolve(opts.hackathon) :
+            this.activeHackathonDataMapper.activeHackathon
+              .pipe(map(hackathon => hackathon.uid))
+              .toPromise()),
+        );
+    }
+    const query = queryBuilder
+      .toString()
+      .concat(';');
+    return from(
+      this.sql.query<number>(query, [], { cache: true }),
+    ).pipe(
+      map((result: number[]) => ({ result: 'Success', data: result[0] })),
+    ).toPromise();
+  }
+
+  // need to be checked for conversion
```

Comment:
What does this mean?

Reply:
This was referred back when I wanted to check if I did the conversion right